styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java

```diff
@@ -330,12 +326,13 @@ public void create(Environment environment) {
         new StateInitializingTrigger(stateFactory, stateManager, storage);
     final TriggerManager triggerManager = new TriggerManager(trigger, time, storage, stats);
 
-    final Scheduler scheduler = new Scheduler(time, timeoutConfig, stateManager, workflowCache,
-                                              storage, trigger);
-
-    final Consumer<Workflow> workflowChangeListener = workflowChanged(workflowCache, storage,
-                                                                      stats, stateManager, time);
+    final WorkflowInitializer workflowInitializer = new WorkflowInitializer(storage, time);
```

Comment:
It would just push down more of the dependencies (storage, time) into the `workflowChanged` handler which does not use them at the moment. So it's a net +1 dependency. I'll keep it here.